flake.nix

```diff
@@ -149,6 +162,8 @@
           rust.cargoManifestPath = "cli/Cargo.toml";
         };
         tools = {
+          # use fenix provided clippy
+          clippy = rust-toolchain.clippy;
```

Comment:
nonblocking: would be nice to fix clippy warnings

Reply:
Are there _new_ warnings?
I can go solve themhere or in a follow up.

Reply:
Yes I presume due to using a different version of clippy

Reply:
Makes sense new rust = smarter clippy
I'll fix those tomorrow then

Reply:
Curious: How did you notice the warnings? Should they block CI?

Reply:
The warnings don't block CI, but they do block pushes locally through git hooks (iff they are in files that were authored on the branch).
other than that, running `cargo clippy` in the `cli` directory will lint the project (speaking of we prob want to add a just target _just_ for that eventually)

Reply:
Warnings won't block pushes unless we enable `settings.clippy.denyWarnings` which is I guess how you were able to push. I think we should enable it since we didn't have any warnings previously and don't want to introduce any more.

Is there a reason for not running the linting in CI? Presumably @mkenigs had to go out of his way to see the warnings and there's no guarantee that all committers will have the hooks configured, which puts quite a lot of onus on reviewers.

Reply:
https://github.com/flox/flox/pull/1375

Reply:
Thanks and thanks ❤️